hyrax/app/assets/javascripts/hyrax/save_work/required_fields.es6

```diff
@@ -7,80 +11,68 @@ export class RequiredFields {
-				}
+  required(elem, label) {
+    $(label).addClass('required control-label')
+		$(label).append('<span class="label label-info required-tag">required</span>')
+		$(elem).addClass('required');
+		$(elem).attr('required', 'required');
+		$(elem).css('box-shadow','0px 0px 0px 2px red');
+  }
 
-				// Remove the required class and span (they will be reset if needed)
-				$(label).removeClass('required')
-				$(label).find('span').remove();
-				
-				// Find the immediate field-wrapper
-				// Gather the inputs
-				this.inputs = $(required).closest('li.field-wrapper').find(':input');
-				// Skip the current elem, any hidden fields and any buttons, skip operator (role), Processed (date) (pre-filled but should not be required)
-				if (this.inputs.filter((n, el) => {
-					if ( 
-						!el.parentNode.className.includes("hidden") && !el.className.includes('btn') && !el.className.includes('remove-hidden') && el != required && el.value !== 'operator' && el.value !== 'Processed' ) {
-						return (el.value !== '');
-					}
-				} ).length > 0) {
-					// Add the required classes, attributes and labels
-					if ( $(label).has('span').length === 0 ) {
-						$(label).addClass('required control-label')
-						$(label).append('<span class="label label-info required-tag">required</span>');
-						$(required).addClass('required');
-						$(required).attr('required', 'required');
-						// Only add the css if the value is empty
-						if ( ($(required).val() === null) || ($(required).val().length < 1) ) {
-							$(required).css('box-shadow','0px 0px 0px 2px red');
-						}
-					}
-				};
-		});
+	isRequiredButBlank(elem) {
+    // if the value is filled in, do not check further
+    if($(elem).hasVal()) return false
 
-		// Find all inputs in multi-nested blocks (where conditional requirements will be in place)
-		this.conditionalRequiredFields = this.form.find('div.multi-nested :input')
-		this.conditionalRequiredFields.change(this.callback)
-	}
+    var requiredButBlank = false
+    if($(elem).data('required')){
+      var requirements = $(elem).data('required').split(';')
+
+      // Special "any" case
+      if(requirements.includes('any')){
+        $(elem).parents('.form-group').find(':input').not('[type=hidden], [data-skip=true]').each(function(i) {
+          if($(this).hasVal()) return (requiredButBlank = true)
+        })
+      } else {
+        // Go through each requirement
+        $.each(requirements, function(i, e) {
+          if($(`input[data-name=${e}]`).hasVal()) return (requiredButBlank = true)
+        })
+      }
 
-	isValuePresent(elem) {
-    return ($(elem).val() === null) || ($(elem).val().length < 1)
+      // TODO this label replace is not right, but ids with [] are not allowed in htlm
+      // and need to be fixed in the nested inputs. Jquery throws an exeption if we try
+      // to match on for block with square brackets
+      var label = $(`label[for=${elem.id.replace(/\[|\]/g, '_')}]`)
```

Comment:
big shout out to the last real todo here - label fors and id matching w/o the []